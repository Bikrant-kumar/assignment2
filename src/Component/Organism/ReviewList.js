import React from 'react';
import ReviewCard from '../Molecules/ReviewCard';
import Grid from '@material-ui/core/Grid';
import Box from '@material-ui/core/Box';
import Label from '../Atoms/Label';

function ReviewList(props) {
    
    return (
        <Box >
            <Label text="REVIEWS"/>
            <Grid container
            item xs={12}>
            {
                props.reviews && 
                props.reviews.map((review) => <ReviewCard review = {review} />)
            }
            </Grid>
        </Box>
    )
}

export default ReviewList