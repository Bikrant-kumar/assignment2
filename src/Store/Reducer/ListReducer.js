export const initState = {    
   
};

export function itemReducer (state = initState, action) {
    switch (action.type) {
        case "DELETE_ITEM":
            const newItem = state.item.filter((item) => item.id !== action.id);
            return{
                ...state,
                item: newItem
            };

        case "DELETE_ROW":
            const newTable= action.table.item.splice(action.table.item.indexOf(action.rowId),1)
                action.table.totalCost = action.table.totalCost - action.rowCost;
                action.table.totalItem= action.table.totalItem- 1;
            return {
                ...state,
                ...newTable
            };
        
        case "ADD_ROW":
            const newRowInTable = action.table.item.push(action.menu)
            action.table.totalCost = action.table.totalCost + action.menu.Cost;
            action.table.totalItem= action.table.totalItem + 1;
            return {
                ...state,
                ...newRowInTable
            };

        case "ADD_ITEM":
            const items = [action.item, ...state.item]
            return {
                ...state,
                item: items,
            };
        case "ADD_REVIEW":
            const review = [action.review,...state.reviews]
            return{
                ...state,
                reviews:review
            };
        
        case "GET_STATE":
            return {  
                ...action.data,
            }

        case "DELETE_ALL_ITEM":
            const newTable1 = action.table.item.splice(0,action.table.item.length)
            action.table.totalCost = 0;
            action.table.totalItem = 0;
            return {
                ...state,
                ...newTable1
            }
            
        default:
            return state;    
    }
};

 export default itemReducer;