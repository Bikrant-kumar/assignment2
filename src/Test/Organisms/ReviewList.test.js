import React from 'react';
import { screen, render } from '@testing-library/react';
import ReviewList from '../../Component/Organism/ReviewList';
import ListReducer from '../../Store/Reducer/ListReducer';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import MockData from '../MockData';
import { createStore } from 'redux';

describe('ReviewList',()=> {

    test('ReviewList Render',() => {   
        const review=
        {
            "id": "review-3",
            "title": "Fish Curry",
            "comments":"Really Good",
            "rating": "3"
        };  
          render(
            <Provider store={createStore(ListReducer, MockData)}>
          <ReviewList review={review} />
          </Provider>);
          expect(screen.getByText("REVIEWS")).toBeInTheDocument();
       });
  
       test('test ReviewList with snapshot',()=>{
         const review=
            {
                "id": "review-3",
            "title": "Fish Curry",
            "comments":"Really Good",
            "rating": "3"
      };  
        const component= renderer.create(
          <Provider store={createStore(ListReducer, MockData)}><ReviewList review = {review}/> </Provider>).toJSON();
        expect(component).toMatchSnapshot();
       });
  
  })