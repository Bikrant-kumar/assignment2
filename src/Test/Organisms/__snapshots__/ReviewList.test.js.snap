// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewList test ReviewList with snapshot 1`] = `
Array [
  <div
    className="MuiBox-root MuiBox-root-3"
  >
    <div
      className="MuiBox-root MuiBox-root-4"
    >
      REVIEWS
    </div>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12"
    />
  </div>,
  " ",
]
`;
